cli/ioctl/cmd/blockchain/blockretrieve.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2019 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package blockchain
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+
+	"github.com/iotexproject/iotex-core/pkg/hash"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	pb "github.com/iotexproject/iotex-core/protogen/iotexapi"
+	pb1 "github.com/iotexproject/iotex-core/protogen/iotextypes"
+)
+
+var serverAddr string
+
+// blockHeaderRetrieveCmd creates a new `ioctl blockchain blockheader` command
+var blockHeaderRetrieveCmd = &cobra.Command{
+	Use:   "blockheader",
+	Short: "Retrieve a block from blockchain",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println(blockHeaderRetrieve(args))
```

Comment:
Not sure if it will print the block header pretty. You may need to define the output format

Reply:
@zjshen14 Thanks, I will incorporate all those review comments, it's WIP yet. I am more concerned about api or api flow to be used for blockheader. I am not able to see it directly. 
Also seems like blockchain.pb.go file is proto version 2.